skate-plugin/src/main/kotlin/com/slack/sgp/intellij/aibot/ChatWindowCompose.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (C) 2024 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.sgp.intellij.aibot
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.defaultMinSize
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.unit.dp
+import org.jetbrains.jewel.foundation.theme.JewelTheme
+import org.jetbrains.jewel.ui.component.DefaultButton
+import org.jetbrains.jewel.ui.component.Text
+import org.jetbrains.jewel.ui.component.TextField
+
+@Composable
+fun ChatWindowCompose(modifier: Modifier = Modifier) {
+  Column(
+    modifier = modifier.fillMaxSize().background(JewelTheme.globalColors.paneBackground),
+    verticalArrangement = Arrangement.Bottom,
+  ) {
+    ConversationField()
+  }
+}
+
+@Composable
+fun ConversationField(modifier: Modifier = Modifier) {
+  var textValue by remember { mutableStateOf(TextFieldValue("")) }
+  Row(modifier) {
+    TextField(
+      value = textValue,
+      onValueChange = { newText -> textValue = newText },
+      modifier = Modifier.weight(1f).padding(4.dp).height(56.dp),
+      placeholder = { Text("Start your conversation") },
+    )
+    Column() {
+      DefaultButton(
+        modifier = Modifier.defaultMinSize(minWidth = 56.dp).padding(4.dp),
+        onClick = { textValue = TextFieldValue("") },
+      ) {
+        Text("Send")
+      }
+      Box(modifier = Modifier.clickable { textValue = TextFieldValue("") }.padding(4.dp)) {
+        Text("Clear Chat")
+      }
```

Comment:
Let's use an `IconButton` instead? Maybe with a trash icon?

Reply:
I couldn't find anything when searching for a trash icon in this link you provided: https://intellij-icons.jetbrains.design/. Instead, I was wondering if we could do a trailing icon that has an "x" button within the chat field. Something like this (instead of clear chat):
<img width="672" alt="Screenshot 2024-08-27 at 3 05 24 PM" src="https://github.com/user-attachments/assets/e2d78fe6-87c5-414e-aa70-f81a8f784510">

Reply:
<img width="278" alt="image" src="https://github.com/user-attachments/assets/f17585ab-2748-4d1d-abe5-de22e3c6a222">


But also I don't think we need one in the text field? That makes sense for a mobile UI where people have a thumb handy and the keyboard is unwieldy, but on desktop it's not a usual pattern. I'd suggest the delete icon to clear the chat and put it next to the "Clear Chat" text

Reply:
I thought it looked a bit off that the Send button had just text and the Clear Chat had an icon, so I added a "send" icon to the send button as well. This way it looks a bit more consistent. What do you think?
<img width="677" alt="Screenshot 2024-08-27 at 4 32 18 PM" src="https://github.com/user-attachments/assets/72dda88b-e31a-408f-a10a-c762b821b6fb">

Reply:
Yeah, maybe just entering to send is easier, but I know DevXP AI and ChatGPT both have send buttons too, so it might be more popular with chatbots? That's also why I wanted to include it since those had it. But also, we can definitely discuss more and adjust in a future PR.

Reply:
hm, ok let's do something in the middle

- Keep the send button but make it disabled unless there's text inputted
- Let's make sure multi-line text and return-to-send work (shift+enter to newline, otherwise enter sends immediately)
- Let's remove the clear text button, it looks like  you have it for the chat text but I would expect this to clear the entire chat above if anything

Reply:
That makes sense, I was able to do the first and third bullet, but I also don't know if Text Field supports shift+enter:
https://github.com/JetBrains/jewel/blob/main/ui/src/main/kotlin/org/jetbrains/jewel/ui/component/TextField.kt

I think maybe I should use a Basic Text Field instead. I will keep you posted. 
<img width="663" alt="Screenshot 2024-08-28 at 12 04 24 PM" src="https://github.com/user-attachments/assets/88a4d5b4-ca56-4acf-ae1b-242abee39318">
<img width="668" alt="Screenshot 2024-08-28 at 12 04 38 PM" src="https://github.com/user-attachments/assets/78a59e4b-12b2-4e2f-a359-eacea0f2befc">

Reply:
I think it should be doable with `Modifier.onKeyEvent`.

Last bit - is the send text redundant here? The other examples I see are icon button only (with a content description)

Reply:
Yeah good point, I can delete the send and add a content description instead. Thanks for the tip!

Reply:
How is this for shift+enter and enter so far?
https://github.com/user-attachments/assets/fbd89bfc-aa76-4c36-b648-9475d4fe4bbd

Reply:
nice!